### PR TITLE
Ignore service principal password resources from Terraform state management

### PR DESCRIPTION
**Pull request by erikschlegel**
## All Submissions:
-------------------------------------
* [YES] Have you followed the guidelines in our Contributing [document](./CONTRIBUTING.md)?
* [YES] Have you added an explanation of what your changes do and why you'd like us to include them?
* [YES] I have updated the documentation accordingly.
* [NA] I have added tests to cover my changes.
* [YES] All new and existing tests passed.
* [YES/NO/NA] My code follows the code style of this project.
* [YES] I ran lint checks locally prior to submission.
* [YES] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

## What is the current behavior?
-------------------------------------
Currently, all azure resource(s) state changes related to the service principal password are managed through Terraform. This results in state changes being triggered for Terraform incremental builds as the password is randomly generated on each run. The impact is Terraform attempts to delete and recreate the password and KV secrets on each incremental deployment, regardless of template changes. Terraform should only be responsible for creating resources like the SP password and keyvault secrets when bootstrapping the initial azure environment. 

Issue Number: #330 

## What is the new behavior?
-------------------------------------
Terraform resource types such as service prinipal passwords and keyvault secrets capturing sp passwords are ignored from state change lifecycles events via 
```
lifecycle {
    ignore_changes = ["value"]
  }
```

## Does this introduce a breaking change?
-------------------------------------
- [NO]

### CODE REVIEW

**Comment by erikschlegel** [comment]
@iphilpot all checks are passing and ready for review

File: infra/modules/providers/azure/service-principal/main.tf [hunk 1]
```diff
@@ -28,4 +28,8 @@ resource "azuread_service_principal_password" "sp" {
   service_principal_id = azuread_service_principal.sp[0].object_id
   value                = local.sp_password
   end_date_relative    = var.sp_pwd_end_date_relative
+
+  lifecycle {
```
Comment (TechnicallyWilliams) [inline infra/modules/providers/azure/service-principal/main.tf]: These lifecycle hooks bake in the original passwords inserted into the creation of the service principal password and the key vault secrets that store them. How will password updates be managed due to this change? Outside of terraform?

Comment (iphilpot) [reply infra/modules/providers/azure/service-principal/main.tf]: Yes should be completed outside of TF

Comment (iphilpot) [inline infra/modules/providers/azure/service-principal/main.tf]: Any reason to put this in the SP module and not in the template?

Comment (iphilpot) [reply infra/modules/providers/azure/service-principal/main.tf]: nevermind, it's a config element it looks like.

**Pull request merged**